nautobot/dcim/views.py

```diff
@@ -4185,6 +4185,51 @@ class VirtualChassisUIViewSet(NautobotUIViewSet):
     table_class = tables.VirtualChassisTable
     queryset = VirtualChassis.objects.all()
 
+    class MembersObjectsTablePanel(object_detail.ObjectsTablePanel):
+        def _get_table_add_url(self, context):
+            obj = get_obj_from_context(context)
+            request = context["request"]
+            return_url = context.get("return_url", obj.get_absolute_url())
+
+            if not request.user.has_perm("dcim.change_virtualchassis"):
+                return None
+
+            params = []
+            master = getattr(obj, "master", None)
+
+            if master:
+                location = getattr(master, "location", None)
+                if location:
+                    params.append(("location", location.pk))
+
+                rack = getattr(master, "rack", None)
+                if rack:
+                    params.append(("rack", rack.pk))
+
+            params.append(("return_url", return_url))
+
+            try:
+                return reverse("dcim:virtualchassis_add_member", kwargs={"pk": obj.pk}) + "?" + urlencode(params)
+            except NoReverseMatch:
+                return None
```

Comment:
What's the case where we'd get a NoReverseMatch here? In other words, is the check even necessary?